controllers/clusterenvironment_controller.go

```diff
@@ -735,7 +748,7 @@ func (r *ClusterEnvironmentReconciler) CreateServiceCatalog(ctx context.Context,
 }
 
 func (r *ClusterEnvironmentReconciler) finalizeClusterEnvironmentInNamespaces(ctx context.Context, ce *primazaiov1alpha1.ClusterEnvironment) error {
-	kcfg, err := clustercontext.GetClusterRESTConfig(ctx, r.Client, ce.Namespace, ce.Spec.ClusterContextSecret)
+	kcfg, err := r.retrieveClusterContextSecret(ctx, ce)
```

Comment:
This is an unrelated change. I made this change based on PR #210
Let me know if I should send a separate PR.